include/networkit/auxiliary/UniformRandomSelector.hpp

```diff
@@ -0,0 +1,52 @@
+#ifndef NETWORKIT_AUXILIARY_UNIFORM_RANDOM_SELECTOR_HPP_
+#define NETWORKIT_AUXILIARY_UNIFORM_RANDOM_SELECTOR_HPP_
+
+#include <cstddef>
+
+#include <networkit/auxiliary/Random.hpp>
+
+namespace Aux {
+
+/**
+ * Select uniformly at random from a set of elements
+ * that is discovered incrementally. Every time you
+ * discover a new element, you tell the selector and
+ * it will tell if you should keep the new element or
+ * your old element. As more and more elements are
+ * discovered, it will be less and less likely that
+ * the new element is chosen in order to ensure that
+ * the selection is uniformly at random. The process
+ * can be reset when a new class of elements is found
+ * that shall be used instead.
+ */
+class UniformRandomSelector {
```

Comment:
Since the elements are discovered incrementally, shouldn't this class include `Incremental` in its name? Just `UniformRandomSelector` looks too generic. Also, this file can be formatted.

Reply:
Renamed to `IncrementalUniformRandomSelector` and added `networkit-format`.